tests/integration/pilot/vm/README.md

```diff
@@ -0,0 +1,88 @@
+# Writing Tests for VMs
+
+Support for including VMs in the integration tests has been added to the framework.
+This document describes the integration and extension mechanisms to exercise VM related code.
+The primary goals are to test VM-related code & ensure it works on a range of supported OS types and versions.
+
+**Note: We currently use mock/simulated VMs for testing purposes. In the future, the testing might switch
+to utilize actual compute instances from different providers.**
+
+## Overview
+
+This section describes the scenarios in which one might want to add a VM test:
+1. Testing existing core Istio features such as traffic management, security, telemetry, etc. for **VMs**
+2. Supporting new OS images for VMs
+3. Testing onboarding tools (iptables, certs, istio-sidecar, etc.) and workflows (services, DNS, etc.) to enmesh a VM
+
+## Secenario 1: Testing VM-related Istio Code
+
+We normally test Istio using Echo application. To test VMs' connectivity, security and telemetry,
+we deploy a regular Echo instance as VM. A VM Echo instance will simulate a VM, disabling kube-dns,
+Service Account mount, etc. For more information around VM onboarding,
+refer to this [doc](https://istio.io/latest/docs/examples/virtual-machines/single-network/).
+
+To deploy an echo instance as a VM
+1. Set the ports for the VMs. Note that due to a bug in WorkloadEntry,
+we need to set the ServicePort to be the same as the InstancePort
+2. Set `DeployAsVm` to be true in `echo.Config`
```

Comment:
For markdown we usually just use 1. for each item and let the render engine figure out the numbering